### PR TITLE
IAP

### PR DESCRIPTION
**Pull request by Nnanyielugo**

### CODE REVIEW

**Comment by mjmayank** [comment]
in case it's relevant, i don't think we would ever have a user with a subscription who doesn't have a zentacle account

File: ios/Configuration.storekit [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          "localizations" : [
+            {
+              "description" : "",
+              "displayName" : "",
+              "locale" : "en_US"
+            }
+          ],
+          "productID" : "pro1gb",
+          "recurringSubscriptionPeriod" : "P1M",
+          "referenceName" : "Pro (1 GB)",
```
Comment (mjmayank) [inline ios/Configuration.storekit:42]: can we change this to drop the (1 GB) from the name? i'm thinking we should just not have it at first and if we want to make products with more than 1 GB or whatever we can add that clarification to those ones

Comment (Nnanyielugo) [reply ios/Configuration.storekit:42]: Need to change it on the AppStore as well, but sure.

File: src/Components/reusables/pro-upsell/ProUpsellDisplay.tsx [hunk 2]
```diff
@@ -94,11 +95,12 @@ const ProUpsellDisplay: FunctionComponent<ProUpsellDisplayProps> = ({
 
   const fetchOfferings = async () => {
     const offerings: PurchasesOfferings = await Purchases.getOfferings();
+    console.log('offerings', offerings);
```
Comment (mjmayank) [inline src/Components/reusables/pro-upsell/ProUpsellDisplay.tsx]: one more console log here

Comment (Nnanyielugo) [reply src/Components/reusables/pro-upsell/ProUpsellDisplay.tsx]: I think the commit after that one removed the console.log statement.

**Review by mjmayank — APPROVED** [review]

File: src/Components/reusables/pro-upsell/ProUpsellDisplay.tsx [hunk 3]
```diff
@@ -71,12 +80,63 @@ const ProUpsellDisplay: FunctionComponent<ProUpsellDisplayProps> = ({
...
   closeText,
 }) => {
+  const [proPackage, setPackage] = React.useState<PurchasesOffering | null>();
+  const [purchaseError, setPurchaseError] = React.useState<string | null>();
+  const [loading, setLoading] = React.useState(false);
+  const dispatch = useAppDispatch();
+
   React.useEffect(() => {
     sendEvent('page_view', {
       screen: 'pro_upsell',
```
Comment (mjmayank) [inline src/Components/reusables/pro-upsell/ProUpsellDisplay.tsx:90]: nit: should be `type: pro_upsell`

File: src/Components/reusables/pro-upsell/ProUpsellDisplay.tsx [hunk 4]
```diff
@@ -71,12 +80,63 @@ const ProUpsellDisplay: FunctionComponent<ProUpsellDisplayProps> = ({
...
+
+      sendEvent('pro__click', {
+        screen: 'pro_upsell',
+      });
+
+      await Purchases.purchasePackage(
+        proPackage?.availablePackages[0] as PurchasesPackage,
+      );
+
+      sendEvent('pro__register', {
```
Comment (mjmayank) [inline src/Components/reusables/pro-upsell/ProUpsellDisplay.tsx:118]: this event should only be sent if the user successfully registers

Comment (Nnanyielugo) [reply src/Components/reusables/pro-upsell/ProUpsellDisplay.tsx:134]: Yeah, that's how it should work. If the register isn't successful, execution should stop there and go to the catch block.

**Pull request merged**